browser/file_persister.go

```diff
@@ -0,0 +1,103 @@
+package browser
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/xk6-browser/env"
+	"github.com/grafana/xk6-browser/storage"
+)
+
+type presignedURLConfig struct {
+	getterURL string
+	headers   map[string]string
+	basePath  string
+}
+
+// newFilePersister will return either a persister that persists file to the local
+// disk or uploads the files to a remote location. This decision depends on whether
+// the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the correct configs.
+func newFilePersister(envLookup env.LookupFunc) (filePersister, error) {
+	envVar, ok := envLookup(env.ScreenshotsOutput)
+	if !ok || envVar == "" {
+		return &storage.LocalFilePersister{}, nil
+	}
+
+	popts, err := parseEnvVar(envVar)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %s: %w", env.ScreenshotsOutput, err)
+	}
+
+	return storage.NewRemoteFilePersister(popts.getterURL, popts.headers, popts.basePath), nil
+}
+
+// parseEnvVar will parse a value such as:
+// url=https://127.0.0.1/,basePath=/screenshots,header.1=a,header.2=b
+// and return them.
+func parseEnvVar(envVarValue string) (presignedURLConfig, error) {
+	ss := strings.Split(envVarValue, ",")
+
+	var (
+		preSignedURLGetterURL string
+		basePath              string
+		headers               = make(map[string]string)
+	)
```

Comment:
Using `presignedURLConfig` here can shorten this code to turn off the linter error 🤔 Adding a `nolint` is fine if it won't work.

Reply:
I feel that `preSignedURLGetterURL` explains what the url is for without any additional comments, so i'd prefer to keep that. I've added a nolint to `parseEnvVar` for now.

Reply:
Fixed in 1790d3e576fcba88ce1e2ff0a38c4c36c34133af

Reply:
Actually I was suggesting this :)

```go
presignedURL := presignedURLConfig{
    headers: make(map[string]string),
}

...

for _, s := range ss {
    ...
    presignedURL.getterURL = v
...
    case "basePath":
        presignedURL.basePath = v
    case "header":
        presignedURL.headers[..]
...
}
return presignedURL, nil
```

Reply:
Right, this feels like a nit pick that doesn't add that much value but i've changed it in 95c63fe0d639f83112969aedf6a1c9c0412cd932